src/canvascontext.ts

```diff
@@ -142,10 +142,15 @@ export class CanvasContext extends RenderContext {
     width = (width / devicePixelRatio) | 0;
     height = (height / devicePixelRatio) | 0;
 
-    canvasElement.width = width * devicePixelRatio;
-    canvasElement.height = height * devicePixelRatio;
-    canvasElement.style.width = width + 'px';
-    canvasElement.style.height = height + 'px';
+    canvas.width = width * devicePixelRatio;
+    canvas.height = height * devicePixelRatio;
+
+    // The canvas could be an instance of either HTMLCanvasElement or an OffscreenCanvas.
+    // Only HTMLCanvasElement has a style attribute.
+    if (canvas instanceof window.HTMLCanvasElement) {
```

Comment:
So in VexFlow 4 we could use `isHTMLCanvas(element)` instead of `instanceof`, to support environments where window is undefined, and cases where the canvas isn't actually an instance of HTMLCanvasElement, but it supplies all the necessary methods and fields.